java/code/src/com/suse/manager/webui/services/SaltServerActionService.java

```diff
@@ -2523,19 +2523,20 @@ public void executeSSHAction(Action action, MinionServer minion, boolean forcePk
                     }, jsonResult -> {
                         String function = (String) call.getPayload().get("fun");
 
-                        // reboot needs special handling in case of ssh push
+                        /* bsc#1197591 ssh push reboot has an response that is not a failure but the action needs to
+                        *  remain pending, in this way SSHPushDriver::getCandidates can schedule a reboot correctly
+                        */
                         if (action.getActionType().equals(ActionFactory.TYPE_REBOOT) &&
-                            sa.getStatus().equals(ActionFactory.STATUS_QUEUED)) {
-                        // if the status is already PICKED_UP, don't change it
-                        // if the status is FAILED or COMPLETED, don't change it
+                                sa.getStatus().equals(ActionFactory.STATUS_QUEUED) ||
+                                sa.getStatus().equals(ActionFactory.STATUS_PICKED_UP)) {
```

Comment:
hmm, I think we had a real problem when the status is already "PICKED_UP" to change the pickup time and the updateServerInfo again. I am not sure if this break things again